### ISSUE TITLE
Receiving empty emails

### ISSUE THREAD

**Issue by CMeldgaard** [verbb/gift-voucher#71]
When the the plugin attatches the voucher pdf, it somehow deletes all the email content. All that i'm receiving is a blank email with an e-mail attatchement.

**Steps to reproduce**
1. Create an order with a voucher purchase
2. Receive empty email with the voucher attatchment

**Additional info**
- Plugin version: 2.6.1
- Craft version: 3.7.13
- Multi-site: Yes

**Additional contex**
I have tried running the function that the plugin triggers on the `EVENT_BEFORE_SEND_MAIL` event. if i remove the line that attatches the pdf, the mail comes with the full content of the email. If i re-introduce the attatch pdf line, i just receive a blank email with pdf's attatched.

**Comment by engram-design** [comment]
I can't seem to replicate this, but I have had reports from time to time of other users having this issue.

Just to be clear, you've tested commenting out this single line in the service:

https://github.com/verbb/gift-voucher/blob/446ac3595821f8544b2e22f946b6ee70508c3d53/src/services/VouchersService.php#L110

And that works as expected (there's email content, but obviously no attachments)?

**Comment by CMeldgaard** [comment]
@engram-design yeah, exactly

**Comment by engram-design** [comment]
@CMeldgaard How odd. So this function comes from SwiftMailer.

I can see in Commerce's own function they attach any files _before_ setting the body content, so I'm wondering if that might be the issue here (we're doing it after, because that's where the event hook is).

As a test, and because I'm unable to replicate it on my end, can you try adding the following after that line?

```
$body = $event->craftEmail->getSwiftMessage()->getBody();
$event->craftEmail->setHtmlBody($body);
$event->craftEmail->setTextBody($body);
```

**Comment by CMeldgaard** [comment]
Hmm, that actually fixes it 🤔 

It's just weird that it's so random who experiences it.

**Comment by engram-design** [comment]
Very interesting! I know SwiftMailer's method of adding content to emails is a bit weird and thinking there's some conflicting there, but glad there's a fix for this.

Fixed in [2.6.2](https://github.com/verbb/gift-voucher/releases/tag/2.6.2)
